test/support/test_config.h

```diff
@@ -221,4 +221,9 @@
 #    define _PSTL_ICPX_FPGA_TEST_USM_VECTOR_ITERATOR_BROKEN 0
 #endif
 
+// A specific kernel compilation order causes incorrect results on Windows with the DPCPP backend. For now, we reorder
+// the test while the issue is being reported to the compiler team. Once it is resolved, this macro can be removed
+// or limited to older compiler versions.
+#define _PSTL_RED_BY_SEG_WINDOWS_COMPILE_ORDER_BROKEN (_MSC_VER && TEST_DPCPP_BACKEND_PRESENT)

```

Comment:
Does the compiler version matter here or does it effect all versions of the compiler?

Reply:
In case if it effect all versions of the compiler, probably better to use compiler version too as newest version of existing compilers. The goal - to recheck it again on newest versions later.

Reply:
Added the compiler version check here. The crash I mentioned with icpx 2024.0 is likely due to outdated drivers on my local Windows setup.

I have used a fresh system with up-to-date drivers and can reproduce with icpx 2024.0 as well.